### ISSUE TITLE
tags not taken into account

### ISSUE THREAD

**Issue by avdv** [etorreborre/specs2#436]
I'm working on a project which uses specs2 version 2.4.9 and I'm trying to upgrade to 3.6.5.

In a mutable spec, examples were tagged like this:

```scala
"A class" should {
    "have foo" in {
      val a = new A()
      a.foo() must not(throwA[Throwable])
    } tag("pool")
}
```

Using specs2 2.4 this works fine and I can "exclude pool" or "include pool" as expected.

But after upgrading to 3.6.5 there's a compiler error:

```
type mismatch;
[error]  found   : String("pool")
[error]  required: org.specs2.data.NamedTag
[error]     } tag("pool")
[error]           ^
[error] one error found
```

Specifying it like this instead:

```scala
tag(org.specs2.data.Tag("pool"))
```

compiles, but it does not work when I run the tests using `testOnly -- exclude pool` as the example / fragment is not excluded.

As a side note, on https://etorreborre.github.io/specs2/guide/SPECS2-3.6.5/org.specs2.guide.Selection.html in the section about the unit specification, it says that

* example 2 is tagged with integration
* example 5 and example 6 are tagged with slow

Both of which seem to be untrue, as I don't even see the words "integration" and "slow" in the example itself.

**Comment by etorreborre** [comment]
Hi Claudio, there is indeed a missing implicit. I am going to add it. In the meantime you can either:

 - move the tag to *before* the example
```
"A class" should {
     // this is the recommended syntax actually
     tag("pool")
    "have foo" in {
      val a = new A()
      a.foo() must not(throwA[Throwable])
    }
}
```
 - use the following  trait
```
import org.specs2.specification.dsl.mutable.TagDsl

trait TagDslFix extends TagDsl { outer =>
  implicit class FragmentTaggedAsString(f: => Fragment) {
    def tag(tag: String) = { outer.tag(tag); f }
    def section(tag: String) = {  outer.tag(tag);  f }
  }
}
```

I will also fix the names in the User Guide, thanks for reporting this.

**Comment by etorreborre** [comment]
BTW I will try to publish a snapshot later on tonight, when I am on a better network.

**Comment by etorreborre** [comment]
I published the jars as [`3.6.5-20151127125638-e36d453`](https://oss.sonatype.org/content/repositories/releases/org/specs2/specs2-core_2.11/3.6.5-20151127125638-e36d453/).

**Comment by avdv** [comment]
Cool, thanks. I'll give it a spin.
